### PR TITLE
Add Additional Metadata Generation During Builds

### PR DESCRIPTION
**Pull request by drasticactions**
Updates the jenkinsFile to include generating [ci-checkout.json](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/dependency-info/7e2421b1775bd41ecd9be9218e6b95c03fb3c7b6/7/package/ci-checkout.json) and [dependency-info.json](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/dependency-info/7e2421b1775bd41ecd9be9218e6b95c03fb3c7b6/7/package/dependency-info.json).

### CODE REVIEW

**Comment by drasticactions** [comment]
> @drasticactions could you elaborate what those 2 generated files do? For those of us who have no clue 😬

`ci-checkout.json` is a list of all of the repositories included in the workspace of the project and their respective hashs.

`dependency-info.json` includes all of the info in `ci-checkout.json`, as well as a scan of the containing `cs/fsproj` files and their containing package and project references.

File: jenkins/Jenkinsfile [hunk 1]
```diff
@@ -482,53 +497,34 @@ timestamps {
                             }
                         }
 
+                        stage ('Generate Workspace Info') {
+                            withCredentials([string(credentialsId: '92cf81e5-6db5-408c-8a0d-192b36200a16', variable: 'GITHUB_AUTH_TOKEN'), string(credentialsId: 'xamarin-build-tasks', variable: 'XAMARIN_BUILD_TASKS')]) {
+					            def redirect = getRedirect()
```
Comment (rolfbjarne) [inline jenkins/Jenkinsfile]: Your indentation is wrong, there seems to be a mix of tabs and spaces here.

File: jenkins/Jenkinsfile [hunk 2]
```diff
@@ -540,7 +536,15 @@ timestamps {
                             uploadFiles ("package/manifest", "jenkins/${branchName}/latest")
 
                             manifestFilename = "manifest"
-                            artifactsFilename = "artifacts.json"
+                        }
+
+                        stage ('Generate artifacts.json') {
+                            withCredentials([string(credentialsId: '92cf81e5-6db5-408c-8a0d-192b36200a16', variable: 'GITHUB_AUTH_TOKEN'), usernamePassword(credentialsId: 'd146d4aa-a437-4633-908f-b455876c44d0', passwordVariable: 'STORAGE_PASSWORD', usernameVariable: 'STORAGE_ACCOUNT')]) {
+					            dir("BuildTasks") {
```
Comment (rolfbjarne) [inline jenkins/Jenkinsfile]: Indentation is off here too.

File: jenkins/Jenkinsfile [hunk 3]
```diff
@@ -482,53 +497,34 @@ timestamps {
                             }
                         }
 
+                        stage ('Generate Workspace Info') {
+                            withCredentials([string(credentialsId: '92cf81e5-6db5-408c-8a0d-192b36200a16', variable: 'GITHUB_AUTH_TOKEN'), string(credentialsId: 'xamarin-build-tasks', variable: 'XAMARIN_BUILD_TASKS')]) {
```
Comment (rolfbjarne) [inline jenkins/Jenkinsfile]: The rest of the file uses space before opening parenthesis (like the mono coding guidelines).

**Comment by drasticactions** [comment]
@rolfbjarne I think I've cleaned up the tabs/spaces inconsistencies, and all of the parentheses as well. I've also uploaded a new build of Xamarin.Build.Tools to address the URLs.

**Comment by rolfbjarne** [comment]
OK, let's see what the build produces now.

**Comment by drasticactions** [comment]
> OK, let's see what the build produces now.

This is the updated [artifacts.json](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/dependency-info/7e2421b1775bd41ecd9be9218e6b95c03fb3c7b6/7/package/artifacts.json) for the build, now including dl.internalx.com links.

**Comment by xamarin-release-manager** [comment]
❌ [Build was (probably) aborted](https://jenkins.internalx.com/job/macios/job/PR-5718/3/display/redirect)

🔥 [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-5718/3/display/redirect) (on internal Jenkins) failed in stage(s) 'Test run' 🔥 : org.jenkinsci.plugins.workflow.steps.FlowInterruptedException

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-5718/3//console)
✅ Packages: [xamarin.ios-12.7.0.214.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-5718/0abb6e8fce29014c2d20b597bb3af8be1c07e8ab/3/package/xamarin.ios-12.7.0.214.pkg) [xamarin.mac-5.7.0.214.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-5718/0abb6e8fce29014c2d20b597bb3af8be1c07e8ab/3/package/xamarin.mac-5.7.0.214.pkg)
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-5718/0abb6e8fce29014c2d20b597bb3af8be1c07e8ab/3/jenkins-results/api-diff/index.html)
ℹ️ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-5718/0abb6e8fce29014c2d20b597bb3af8be1c07e8ab/3/jenkins-results/apicomparison/api-diff.html) (please review changes)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-5718/0abb6e8fce29014c2d20b597bb3af8be1c07e8ab/3/jenkins-results/generator-diff/index.html) (no change)
🔥 [Failed to publish results](https://jenkins.internalx.com/job/macios/job/PR-5718/3//console) 🔥

**Comment by xamarin-release-manager** [comment]
❌ [Build was (probably) aborted](https://jenkins.internalx.com/job/macios/job/PR-5718/4/display/redirect)

🔥 [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-5718/4/display/redirect) (on internal Jenkins) failed in stage(s) 'Test run, Test run' 🔥

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-5718/4//console)
✅ Packages: [xamarin.ios-12.7.0.215.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-5718/d6c39742572d13f32d5f45a92e706804d13e5998/4/package/xamarin.ios-12.7.0.215.pkg) [xamarin.mac-5.7.0.215.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-5718/d6c39742572d13f32d5f45a92e706804d13e5998/4/package/xamarin.mac-5.7.0.215.pkg)
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-5718/d6c39742572d13f32d5f45a92e706804d13e5998/4/jenkins-results/api-diff/index.html)
ℹ️ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-5718/d6c39742572d13f32d5f45a92e706804d13e5998/4/jenkins-results/apicomparison/api-diff.html) (please review changes)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-5718/d6c39742572d13f32d5f45a92e706804d13e5998/4/jenkins-results/generator-diff/index.html) (no change)
🔥 [Test run failed](http://xamarin-storage/jenkins/xamarin-macios/PR-5718/d6c39742572d13f32d5f45a92e706804d13e5998/4/jenkins-results/tests/index.html) 🔥

# Test results

<details>
<summary>7 tests failed, 0 tests skipped, 341 tests passed.</summary>

## Failed tests

 * monotouch-test/tvOS - simulator/Debug: BuildFailure
 * framework-test/tvOS - simulator/Debug: BuildFailure
 * monotouch-test/tvOS - simulator/Debug (static registrar): BuildFailure
 * monotouch-test/tvOS - simulator/Release (all optimizations): BuildFailure
 * monotouch-test/tvOS - simulator/Debug (all optimizations): BuildFailure
 * MSBuild tests/iOS: Failed (Execution failed with exit code 27)
 * MTouch tests/NUnit: Failed (Execution failed with exit code 6)
</details>

**Comment by rolfbjarne** [comment]
Test failures are unrelated (all due to https://github.com/xamarin/maccore/issues/1465).

**Pull request merged**